pilot/pkg/xds/filters/filters.go

```diff
@@ -164,6 +164,13 @@ var (
 
 	HTTPMx = buildHTTPMxFilter()
 
+	IstioNetworkAuthenticationFilter = &listener.Filter{
```

Comment:
One concern - this will apply to all Envoy connected, but its only going to be in 1.17 proxies. So we may need to guard all the changes around `IsIstioVersionGE117` (new function based on `IsIstioVersionGE116`)?

Reply:
Valid concern, added guards for earlier version.